### ISSUE TITLE
Prometheus Operator fails to create the kubelet service and endpoints

### ISSUE THREAD

**Issue by jungopro** [coreos/prometheus-operator#1998]
**What did you do?**

Deployed prometheus-operator using the helm chart

**What did you expect to see?**

kubelet service and associated endpoints created in the kube-system namespace

**What did you see instead? Under which circumstances?**

Neither service nor endpoints got created

**Environment**

* Prometheus Operator version:

	v.0.20.0

* Kubernetes version information:

	1.9.0

* Kubernetes cluster kind: 

kubespray

* Manifests:

```
insert manifests relevant to the issue
```

* Prometheus Operator Logs:

```
level=error ts=... caller=operator.go:378 component=prometheusoperator msg="syncing nodes into Endpoints object failed" err="failed to determine hostname for node (ac-medium-worker-3): host address unknown"
```

- I have verified that ```--kubelet-service``` arg is passed to the pod properly according to [this document](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/cluster-monitoring.md#preparing-kubernetes-components) and tried resetting the pod, etc.
- I don't see any problem with name resolution in the cluster that I could find... (maybe I missed something?)

Appreciate any help
Omer

**Comment by brancz** [comment]
Yes this exact issue does not block anymore on latest versions :)
